buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/LicenseHeadersTask.groovy

```diff
@@ -33,8 +36,15 @@ import java.nio.file.Files
  */
 public class LicenseHeadersTask extends AntTask {
 
+    // checks all the java sources (allJava)
+    private Iterable<File> input = project.sourceSets.collect({it.allJava.srcDirs}).flatten()
```

Comment:
But still do that in an afterEvaluate, since sourceSets can be added (eg we do this in standalone test base class).